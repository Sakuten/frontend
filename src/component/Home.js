import styled from 'styled-components'
import React from 'react'
import {Link} from 'react-router-dom'
import cloud from '../cloud.svg'
import week from '../week.svg'
import sousaku from '../sousaku.svg'

const UpperHeading = styled.div`
  position: relative;
  height: ${props => props.theme.header_height}rem;
  background-color: ${props => props.theme.header_color_upper};
  color: white;
`

const LowerHeading = styled.div`
  position: relative;
  height: ${props => props.theme.header_height}rem;
  background-color: ${props => props.theme.header_color_lower};
  color: white;
`

const Title = styled.img`
  position: absolute;
  right: 5%;
  bottom: 20%;
  z-index: 10;

  height: ${props => props.theme.header_height - 2.5}rem;
`

const SubTitle = styled.img`
  position: absolute;
  right: 30%;
  bottom: 20%;

  height: ${props => props.theme.header_height - 2.5}rem;
`

const Button = styled.div`
  margin: 0 32px;
  background-color: #00cec9;
`

const ButtonText = styled.div`
  padding: 30px;
  text-align: center;
  color: white;
`

const Cloud = styled.div`
  position: absolute;
  width: 40vw;
  left: -10%;
  top: -60%;
`

const Container = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 80vh;
`

const Home = () => (
  <div>
    <UpperHeading>
      <Title src={week} />
    </UpperHeading>
    <LowerHeading>
      <Cloud>
        <img src={cloud} />
      </Cloud>
      <SubTitle src={sousaku} />
    </LowerHeading>
    <Container>
      <Button>
        <Link data-test='home-login' to='/lottery/login'>
          <ButtonText>
            <h3>QRコード読み取り</h3>
            <p>(カメラが起動します)</p>
          </ButtonText>
        </Link>
      </Button>
    </Container>
  </div>
)

export default Home
