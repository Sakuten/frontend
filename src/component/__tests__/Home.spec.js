import React from 'react'
import { shallow } from 'enzyme'
import Home from '../Home'

const setup = propOverrides => {
  const props = propOverrides

  const wrapper = shallow(<Home {...props} />)

  return {
    props,
    wrapper,
    link: wrapper.find('[data-test="home-login"]')
  }
}

describe('components', () => {
  describe('Home', () => {
    it('render', () => {
      const { wrapper } = setup()
      expect(wrapper).toMatchSnapshot()
    })

    it('renders login link', () => {
      const { link } = setup()
      expect(link.length).toBe(1)
    })
  })
})
