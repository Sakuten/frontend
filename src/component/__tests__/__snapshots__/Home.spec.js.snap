// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Home render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.header>
          <img
            src="header_2019.png"
          />
        </styled.header>,
        <styled.div>
          <Unknown
            data-test="home-login"
            to={
              Object {
                "pathname": "/lottery/login",
                "search": undefined,
              }
            }
          >
            <h3>
              QRコード読み取り
            </h3>
            <p>
              (カメラが起動します)
            </p>
          </Unknown>
        </styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <img
            src="header_2019.png"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "header_2019.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Unknown
            data-test="home-login"
            to={
              Object {
                "pathname": "/lottery/login",
                "search": undefined,
              }
            }
          >
            <h3>
              QRコード読み取り
            </h3>
            <p>
              (カメラが起動します)
            </p>
          </Unknown>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <h3>
                QRコード読み取り
              </h3>,
              <p>
                (カメラが起動します)
              </p>,
            ],
            "data-test": "home-login",
            "to": Object {
              "pathname": "/lottery/login",
              "search": undefined,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "QRコード読み取り",
              },
              "ref": null,
              "rendered": "QRコード読み取り",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "(カメラが起動します)",
              },
              "ref": null,
              "rendered": "(カメラが起動します)",
              "type": "p",
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.header>
            <img
              src="header_2019.png"
            />
          </styled.header>,
          <styled.div>
            <Unknown
              data-test="home-login"
              to={
                Object {
                  "pathname": "/lottery/login",
                  "search": undefined,
                }
              }
            >
              <h3>
                QRコード読み取り
              </h3>
              <p>
                (カメラが起動します)
              </p>
            </Unknown>
          </styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <img
              src="header_2019.png"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "src": "header_2019.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Unknown
              data-test="home-login"
              to={
                Object {
                  "pathname": "/lottery/login",
                  "search": undefined,
                }
              }
            >
              <h3>
                QRコード読み取り
              </h3>
              <p>
                (カメラが起動します)
              </p>
            </Unknown>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <h3>
                  QRコード読み取り
                </h3>,
                <p>
                  (カメラが起動します)
                </p>,
              ],
              "data-test": "home-login",
              "to": Object {
                "pathname": "/lottery/login",
                "search": undefined,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "QRコード読み取り",
                },
                "ref": null,
                "rendered": "QRコード読み取り",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "(カメラが起動します)",
                },
                "ref": null,
                "rendered": "(カメラが起動します)",
                "type": "p",
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
